File: task3_python/Weapon.py
import abc


class Weapon(object):
	__metaclass__ = abc.ABCMeta
	def __init__(self,ranges,damage,owner):
		self.ranges = ranges
		self.effect = damage
		self.owner = owner 

	@abc.abstractmethod
	def action(posx,posy):
		return

	@abc.abstractmethod
	def enhance(self):
		return

	

File: part1/Weapon.java
public abstract class Weapon {
	protected final int range;
	protected int effect;
	protected Player owner;
	
	protected Weapon(int range, int damage, Player owner) {
		this.range = range;
		this.effect = damage;
		this.owner = owner;
	}
	
	abstract public void action(int posx, int posy);
	abstract public void enhance();

	public int getEffect() {
		return this.effect;
	}

	public int getRange() {
		return this.range;
	}
}
